adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -1301,6 +1304,29 @@
             return promise.promise;
         }
 
+        function checkIfRealmAvailable() {
+            return new Promise(function (resolve, reject) {
+                var url = getRealmUrl();
+                var req = new XMLHttpRequest();
+                req.open('GET', url, true);
+                req.setRequestHeader('Accept', 'application/json');
+                req.onload = function () {
+                    try {
+                        var res = JSON.parse(req.responseText);
+                        if (res.realm != kc.realm)
+                            reject(new Error('Invalid response from Keycloak realm'));
```

Comment:
Using Error object looks good to me and correct. However, we are not using it anywhere else. I think there may be some existing code that can be broken when we introduce this as it can expect errorData directly. Maybe we should not use Error now, but introduce enhancement Jira and replace it everywhere. WDYT?

Reply:
You right. Error is canonical way to represent errors in JS, but as declared in TypeScript API (keycloak.d.ts), in case of error we should return KeycloakError. So lets respect this behavior.